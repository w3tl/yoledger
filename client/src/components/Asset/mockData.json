[{
  "__typename": "Account",
  "id": "1",
  "name": "Bank Card",
  "balance": 0
}, {
  "__typename": "Account",
  "id": "2",
  "name": "Cash",
  "balance": 10
}, {
  "__typename": "Account",
  "id": "3",
  "name": "Bank Card 2",
  "balance": 150
}]
