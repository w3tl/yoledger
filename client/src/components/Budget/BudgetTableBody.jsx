import React from 'react';
import PropTypes from 'prop-types';
import BudgetTableRow from './BudgetTableRow';
import { budgetPropType } from './propTypes';
import withData from './BudgetTableBodyHOC';

class BudgetTableBody extends React.Component {
  static propTypes = {
    accounts: PropTypes.arrayOf(budgetPropType),
    onUpdate: PropTypes.func,
  }

  static defaultProps = {
    accounts: [],
    onUpdate: () => {},
  }

  handleChange = account => ({ date, amount }) => {
    const { onUpdate } = this.props;
    const input = { account, date, amount };
    onUpdate({ variables: { input } });
  }

  render() {
    const { accounts } = this.props;
    return (
      <div>
        {accounts.map(budget => (
          <BudgetTableRow
            key={budget.account.name}
            onChange={this.handleChange(budget.account.name)}
            budget={budget}
          />
        ))}
      </div>
    );
  }
}

export { BudgetTableBody as BudgetTableBodyComponent };
export default withData(BudgetTableBody);
