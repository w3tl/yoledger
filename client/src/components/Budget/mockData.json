{
  "__typename": "BudgetPlan",
  "periods": ["2018-06-10", "2018-06-25", "2018-07-10", "2018-07-25", "2018-08-10", "2018-08-25"],
  "accounts": [{
    "account": {
      "__typename": "Account",
      "id": "1",
      "name": "Food"
    },
    "allocations": [{
      "__typename": "Allocation",
      "date": "2018-06-10",
      "amount": 10,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-06-25",
      "amount": 12,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-10",
      "amount": 14,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-07-25",
      "amount": 18,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-10",
      "amount": 7,
      "balance": 0
    }, {
      "__typename": "Allocation",
      "date": "2018-08-25",
      "amount": 6,
      "balance": 0
    }]
  }]
}
