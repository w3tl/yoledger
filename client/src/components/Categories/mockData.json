{
  "expenses": [{
    "id": "21",
    "name": "Food",
    "type": "EXPENSE"
  }, {
    "id": "22",
    "name": "Beer",
    "type": "EXPENSE"
  }, {
    "id": "23",
    "name": "Train",
    "type": "EXPENSE"
  }],
  "incomes": [{
    "id": "31",
    "name": "Salary",
    "type": "INCOME"
  }, {
    "id": "32",
    "name": "Debts",
    "type": "INCOME"
  }, {
    "id": "33",
    "name": "Cashback",
    "type": "INCOME"
  }]
}
