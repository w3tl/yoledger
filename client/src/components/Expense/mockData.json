[{
  "id": "21",
  "name": "Food"
}, {
  "id": "22",
  "name": "Beer"
}, {
  "id": "23",
  "name": "Train"
}]
