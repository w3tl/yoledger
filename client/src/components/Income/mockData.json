[{
  "id": "31",
  "name": "Salary"
}, {
  "id": "32",
  "name": "Debts"
}, {
  "id": "33",
  "name": "Dividends"
}]
