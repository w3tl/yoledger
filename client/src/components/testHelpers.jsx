import React from 'react';
import { shallow, mount } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

export const wait = (ms = 0) => new Promise(resolve => setTimeout(resolve, ms));

export const withProvider = Component => props => (
  <MockedProvider {...props}>
    <Component />
  </MockedProvider>
);

export const testRenderWithoutError = (Component) => {
  it('renders without error', async () => {
    const wrapper = shallow(Component);
    await wait();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
};

export const testLoadingState = (Component) => {
  it('should render loading state initially', () => {
    const wrapper = mount(Component);
    expect(wrapper.text()).toEqual('Loading...');
  });
};

export const testErrorUI = (Component) => {
  it('should show error UI', async () => {
    const wrapper = mount(Component);
    await wait(0); // wait for response
    expect(wrapper.update().text()).toContain('Error');
  });
};

// THNX: http://blog.blakesimpson.co.uk/read/98-how-to-stub-date-in-jest-tests-building-a-stubdate-helper
export const stubDate = (fixedDate) => {
  let originalDate;

  beforeAll(() => {
    originalDate = Date;

    global.Date = class extends Date {
      constructor() {
        super();

        return fixedDate;
      }
    };
  });

  afterAll(() => {
    global.Date = originalDate;
  });
};
