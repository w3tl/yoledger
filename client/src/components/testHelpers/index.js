import React from 'react';
import { shallow, mount } from 'enzyme';
import { ApolloProvider } from 'react-apollo';
import { ApolloClient } from 'apollo-client';
import { InMemoryCache } from 'apollo-cache-inmemory';
import { SchemaLink } from 'apollo-link-schema';
import { makeExecutableSchema } from 'graphql-tools';
import typeDefs from './typeDefs';
import resolvers from './resolvers';

export const wait = (ms = 0) => new Promise(resolve => setTimeout(resolve, ms));

const executableSchema = makeExecutableSchema({
  typeDefs,
  resolvers,
});

const cache = new InMemoryCache();
const link = new SchemaLink({ schema: executableSchema });
export const client = new ApolloClient({
  link,
  cache,
});

export const withProvider = Component => props => (
  <ApolloProvider client={client}>
    <Component client={client} />
  </ApolloProvider>
);

export const testRenderWithoutError = (Component) => {
  it('renders without error', async () => {
    const wrapper = shallow(Component);
    await wait();
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
  });
};

export const testLoadingState = (Component) => {
  it('should render loading state initially', () => {
    const wrapper = mount(Component);
    expect(wrapper.text()).toEqual('Loading...');
  });
};

export const testErrorUI = (Component) => {
  it('should show error UI', async () => {
    const wrapper = mount(Component);
    await wait(0); // wait for response
    expect(wrapper.update().text()).toContain('Error');
  });
};

// THNX: http://blog.blakesimpson.co.uk/read/98-how-to-stub-date-in-jest-tests-building-a-stubdate-helper
export const stubDate = (fixedDate) => {
  const date = new Date(fixedDate);
  let originalDate;

  beforeAll(() => {
    originalDate = Date;

    global.Date = class extends Date {
      constructor() {
        super();

        return date;
      }
    };
  });

  afterAll(() => {
    global.Date = originalDate;
  });
};
