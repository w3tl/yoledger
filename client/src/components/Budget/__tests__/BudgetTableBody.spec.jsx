import React from 'react';
import { mount } from 'enzyme';
import { BudgetTableBodyComponent } from '../BudgetTableBody';
import { testRenderWithoutError } from '../../testHelpers';
import { accounts } from '../mockData';

describe('BudgetTableBody', () => {
  testRenderWithoutError(<BudgetTableBodyComponent accounts={accounts} onUpdate={() => {}} />);

  it('should call onUpdate with variables', () => {
    const onUpdate = jest.fn();
    const output = mount(
      <BudgetTableBodyComponent accounts={accounts} onUpdate={onUpdate} />,
    );

    output.find('input').first().simulate('blur');
    expect(onUpdate.mock.calls[0]).toMatchSnapshot();
  });
});
