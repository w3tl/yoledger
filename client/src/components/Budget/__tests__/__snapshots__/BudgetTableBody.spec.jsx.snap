// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetTableBody renders without error 1`] = `
<div>
  <BudgetTableRow
    budget={
      Object {
        "account": Object {
          "__typename": "Account",
          "id": "1",
          "name": "Food",
        },
        "allocations": Array [
          Object {
            "__typename": "Allocation",
            "amount": 10,
            "balance": 0,
            "date": "2018-06-10",
          },
          Object {
            "__typename": "Allocation",
            "amount": 12,
            "balance": 0,
            "date": "2018-06-25",
          },
          Object {
            "__typename": "Allocation",
            "amount": 14,
            "balance": 0,
            "date": "2018-07-10",
          },
          Object {
            "__typename": "Allocation",
            "amount": 18,
            "balance": 0,
            "date": "2018-07-25",
          },
          Object {
            "__typename": "Allocation",
            "amount": 7,
            "balance": 0,
            "date": "2018-08-10",
          },
          Object {
            "__typename": "Allocation",
            "amount": 6,
            "balance": 0,
            "date": "2018-08-25",
          },
        ],
      }
    }
    key="Food"
    onChange={[Function]}
  />
</div>
`;

exports[`BudgetTableBody should call onUpdate with variables 1`] = `
Array [
  Object {
    "variables": Object {
      "input": Object {
        "account": "Food",
        "amount": 10,
        "date": "2018-06-10",
      },
    },
  },
]
`;
